core/src/main/scala/scalaz/EitherT.scala

```diff
@@ -287,6 +287,39 @@ object EitherT extends EitherTInstances {
 
 }
 
+object EitherTNondeterminism {
+  implicit def eitherTNondeterminism[F[_], L](implicit F0: Nondeterminism[F]): Nondeterminism[EitherT[F, L, ?]] =
+    new Nondeterminism[EitherT[F, L, ?]] {
+      val F = Nondeterminism[F]
+
+      override def chooseAny[A](
+                                 head: DisjunctionT[F, L, A],
+                                 tail: Seq[DisjunctionT[F, L, A]]): DisjunctionT[F, L, (A, Seq[DisjunctionT[F, L, A]])] =
+        new EitherT(F.map(F.chooseAny(head.run, tail map (_.run))) {
+          case (a, residuals) =>
+            a.map((_, residuals.map(new EitherT(_))))
+        })
```

Comment:
Add a test to make sure that, after emptying a sequence of `EitherT`s with `chooseAny`, you have sequenced each `F` effect only once.